### PR TITLE
fix: braking while taxiing should not turn autobrake off

### LINKED ISSUE #255: Braking while taxiing turn autobrake off
If you break for traffic while taxiing autobrakes are turned off.

### PR DESCRIPTION
**Pull request by geoffda**
Fixes #255

## Summary of Changes

The initial bug has been fixed so that braking when taxiing while autobrakes are armed does not disarm them. Additionally, the disarming of autobrake pushbutton switches is now as close to real life as I can make it.
1) Autobrakes will be disarmed approximately 10 seconds after landing gear is retracted.
2) Autobrake MAX cannot be set while the plane is in the air.
3) If autobrakes activate upon landing, we will automatically disarm the switch once the autobrakes deactivate.

## Screenshots (if necessary)

No visual changes.

## References

http://a320dp.com/A320_DP/landing-gear/sys-12.8.3.html

## Additional context

Discord username (if different from GitHub):
geoffda

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created and uploaded.
The build script will have already been run with the latest changes, so no need to rerun it once you download the zip.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the right side, click on the **Artifacts** drop down and click the **A32NX** link

### CODE REVIEW

File: A32NX/ModelBehaviorDefs/A32NX/Interior/A32NX_Interior_Misc.xml [hunk 1]
```diff
@@ -109,7 +109,7 @@
 
     Takes into account the value of L:XMLVAR_SWITCH_OVHD_INTLT_ANNLT_Position.
 
-    Authors: Asobo / Markus Ullmann (@jokey2k) 18/10/20
+    Authors: Asobo / Markus Ullmann (@jokey2k) 18/10/20 Geoff Darst(geoffda) 2/11/20
```
Comment (MisterChocker) [inline A32NX/ModelBehaviorDefs/A32NX/Interior/A32NX_Interior_Misc.xml:112]: I could be wrong, but use the `changelog.md` instead

**Review by MisterChocker — CHANGES_REQUESTED** [review]

**Comment by MisterChocker** [comment]
don't forget to add changes to changelog.md

**Comment by lukecologne** [comment]
You don't need to add the author to the templates, that information can be seen via git blame.

**Comment by geoffda** [comment]
> don't forget to add changes to changelog.md

Thanks, I'm assuming you mean the one in root? Is there a master list for the [Changed Area] somewhere?

**Comment by lukecologne** [comment]
https://github.com/flybywiresim/a32nx/blob/master/.github/CHANGELOG.md

**Comment by geoffda** [comment]
> You don't need to add the author to the templates, that information can be seen via git blame.

Thanks. So in terms of implementing feedback, how does that work? Do I fix things locally and then do a new pull request?

**Comment by lukecologne** [comment]
No, you can just push new commits to your branch

**Comment by geoffda** [comment]
> No, you can just push new commits to your branch

Cool. Ok, I just pushed the changes to remove my name from the template, as well as the changelog.md up.

File: A32NX/ModelBehaviorDefs/A32NX/Interior/A32NX_Interior_Misc.xml (hunk 1 above)
Comment (geoffda) [reply A32NX/ModelBehaviorDefs/A32NX/Interior/A32NX_Interior_Misc.xml]: Done.

**Comment by ImenesFBW** [comment]
Thanks for this, I was waiting impatiently for this improvement hence I tested it a bit even though it is not tagged ready for test yet. The functionality is working as described. I had one question about pressing autobreak MAX  in the air. Now when you press the autobreak MAX in the air it clears your previous selection. I think in the real thing it should not clear your selection when you press MAX , but even some sims like LEVEL-D don't simulate that behavior correctly. Video below!

Thank you very much again for this great work, can't wait to see it in the master! I will write a full QA template comment when this gets tagged as ready for test.

https://streamable.com/g7ccm2

**Comment by geoffda** [comment]
> Thanks for this, I was waiting impatiently for this improvement hence I tested it a bit even though it is not tagged ready for test yet. The functionality is working as described. I had one question about pressing autobreak MAX in the air. Now when you press the autobreak MAX in the air it clears your previous selection. I think in the real thing it should not clear your selection when you press MAX , but even some sims like LEVEL-D don't simulate that behavior correctly. Video below!
> 
> Thank you very much again for this great work, can't wait to see it in the master! I will write a full QA template comment when this gets tagged as ready for test.
> 
> https://streamable.com/g7ccm2

Thank you. Good catch! This is any easy change to make. I'm going to test it now. Should I just push this into the existing pull request (assuming it works as expected)?

**Comment by MisterChocker** [comment]
> Thank you. Good catch! This is any easy change to make. I'm going to test it now. Should I just push this into the existing pull request (assuming it works as expected)?

I always try to ensure when pushing to an PR that all my changes are working, except I do some super simple stuff like changing some sort of text, string whatever. But that's my approach.

**Comment by Snapmatics** [comment]
MAX Auto Brake didn't disengage when I was trying to abort my take off. Once Take Off Power is set it changes how it works and it can be disengaged.

**Comment by geoffda** [comment]
> MAX Auto Brake didn't disengage when I was trying to abort my take off. Once Take Off Power is set it changes how it works and it can be disengaged.

Can you provide details on what should happen in that situation? I suspect that this would need to be fixed within the autobrake system itself (rather than just the switch behavior).

**Comment by danbw9** [comment]
> Can you provide details on what should happen in that situation? I suspect that this would need to be fixed within the autobrake system itself (rather than just the switch behavior).

I discussed with @Snapmatics and saw what was happening. When aborting a take off, if AUTO BRK MAX is selected prior to take off, it will activate and should a green 'DECEL' light when conditions have been met.  Then when pilot applies pressure on the brakes, the AUTO BRK should disengage in that phase like on landing. I guess the aircraft does not know exactly what phase of flight it is in and that needs to be defined.

**Comment by geoffda** [comment]
Found one issue, that I've got a fix for and am retesting. After landing, autobrakes can't be reset. My fault: I had tested that scenario early on, but forgot to test again when I made the change that broke it.

**Comment by geoffda** [comment]
Is there anything else I need to do at this point beyond keeping the pull request in sync with master?

**Comment by Benjozork** [comment]
Nope, LGTM

**Review by Benjozork — APPROVED** [review]

**Comment by flightsimsimon** [comment]
Name: Discord User Flightsimsimon#1370
Date of testing: Current date 08/11/2020
Version of the sim: 10.7.1
PR Tested: #1762 
Object of test: (ECAM Page or Flight model or views or anything, what the PR modifies)

Testing technique: Description of how you tested the Pull Request
Full cold and dark start to take off
Test streamed, view stream here: https://youtu.be/_vOdl_QX_K0

Steps you've made to test the Pull Request:
1. Call and dark start up
2. taxi
3. take off
4.
5.

Flight notes

Startup: Notes on the startup (Everything is normal, or I noticed something weird)
Autobreak selected in set up

Taxi/Takeoff : Notes on the taxi an takeoff (Everything is normal, or this time the climb rate seems bigger than usual)
Worked perfectly, did not disengage when clicked
Disconnected
 10 seconds after gear up as expected

Cruise: Notes on cruise (Everything is normal, or I noticed something)
Worked perfect, could not select in flight as per the SOP

Descent/Landing : Notes on descent and landing (Work as expected or this time the descent rate seems bigger than usual)
NOT TESTED

Shutdown: Notes on the shutdown of the aircraft
NOT TESTED

Issues (If Any): 
Overall Rating: Good/To be improved
Conclusions of the testing: Make a small descriptive to areas where it needs to be improved and recommendations.

Worked perfectly - OK to merge 

FinalLight-NL, z0rgmeister, SmallLackey

**Comment by ChruutvoLuzi** [comment]
Name: ChruutvoLuzi | SD-PB.ch#8902
Date of testing: Current date 8.11.2020
Version of the sim: 1.8.10.0
PR Tested: #1762 
Object of test: Fix park brake disabling while braking

Testing technique: Try out
Steps you've made to test the Pull Request:
1. Started cold and dark
2. turned on the aircraft as usual
3. took off
4. approached and landed
5. shutdown the aircraft

Flight notes

Startup: n/a

Taxi/Takeoff : Parking brake does not disengage when braking with the toe brake. Autobrake Max is not selectable aswell. Autobrake turned off as soon as the gear was retracted

Cruise: n/a

Descent/Landing : Landed with Autobrake low. Autobrake automatically disengaged after decceleration

Shutdown: n/a

Issues (If Any): 
Overall Rating: Good
Conclusions of the testing: Works as described

**Comment by danbw9** [comment]
I was just testing this as it got merged. 

All good apart from that I found a small bug with that on landing after intervening with manual braking, the AUTO BRK (LOW or MED in blue) indication is still there. 

I am surprised no one spotted this but I guess no one really checks the AUTO BRK indication after disengaging it.

**Comment by geoffda** [comment]
I tested this and it works. Most likely, you forgot to arm the ground spoilers so the auto brakes never engaged. In that case, they will remain armed.

**Comment by danbw9** [comment]
> I tested this and it works. Most likely, you forgot to arm the ground spoilers so the auto brakes never engaged. In that case, they will remain armed.

I landed with GND spoilers armed. landing no blue. 

What you say is correct however connected with another bug. When I cancel REV Thrust I move the throttles slightly above idle then back to idle detent. This automatically puts the SPD BRK lever in the disarm position which is not correct (as the PF usually disarms it manually). If I intervene with manual braking after this point - this is where I saw the AUTO BRK indication still illuminated, after AUTO BRK had been active already before.

**Comment by geoffda** [comment]
Thanks for confirming. Sounds like another issue that should get addressed with the rewrite of the autobraking system.

Sent from Mail for Windows 10

**Pull request merged**